ocs_ci/helpers/dr_helpers.py

```
@@ -562,213 +561,264 @@ def check_mirroring_status_for_custom_pool(
-        namespace (str): the namespace of the VRG resources
-        resource_name (str): Name of VRG resource
+        kind (str): Kind of resource (e.g., constants.VOLUME_REPLICATION, constants.REPLICATION_SOURCE, etc.)
+        namespace (str): The namespace of the resources
+        expected_count (int): Expected number of resources
+        timeout (int): Time in seconds to wait
 
-    Returns:
-        bool: True if VRG is in expected state or was deleted, False otherwise
+    Raises:
+        TimeoutExpiredError: If expected number of resources not reached
 
     """
+    if expected_count == 0:
+        logger.info(f"Waiting for all {kind} to be deleted")
+    else:
+        logger.info(f"Waiting for {expected_count} {kind} to be created")
 
-    vrg_obj = ocp.OCP(
-        kind=constants.VOLUME_REPLICATION_GROUP,
+    sample = TimeoutSampler(
+        timeout=timeout,
+        sleep=5,
+        func=get_resource_count,
+        kind=kind,
         namespace=namespace,
     )
-    if resource_name:
-        vrg_list = vrg_obj.get(resource_name=resource_name)
-        vrg_list_index = vrg_list
-    else:
-        vrg_list = vrg_obj.get().get("items")
-        vrg_list_index = vrg_list[0]
-
-    # Skip state check if resource was deleted
-    if len(vrg_list) == 0 and state.lower() == "secondary":
-        ocs_version = version.get_semantic_ocs_version_from_config()
-        if ocs_version <= version.VERSION_4_17:
-            logger.info("VRG resource not found, skipping state check")
-            return True
-        else:
-            logger.info("VRG resource not found")
-            return False
-    vrg_name = vrg_list_index["metadata"]["name"]
-    desired_state = vrg_list_index["spec"]["replicationState"]
-    current_state = vrg_list_index["status"]["state"]
-    logger.info(
-        f"VRG: {vrg_name} desired state is {desired_state}, current state is {current_state}"
+    sample.wait_for_func_value(expected_count)
```

Need to raise an Exception